modules/consensus/synchronize.go

```diff
@@ -170,16 +170,11 @@ func (cs *ConsensusSet) managedReceiveBlocks(conn modules.PeerConn) (returnErr e
 			// The last block received will be the current block since
 			// managedAcceptBlock only returns nil if a block extends the longest chain.
 			currentBlock := cs.managedCurrentBlock()
-			// COMPATv0.5.1 - broadcast the block to all peers <= v0.5.1 and block header to all peers > v0.5.1
-			var relayBlockPeers, relayHeaderPeers []modules.Peer
+			// broadcast the block header to all peers
+			var relayHeaderPeers []modules.Peer
 			for _, p := range cs.gateway.Peers() {
-				if build.VersionCmp(p.Version, "0.5.1") <= 0 {
-					relayBlockPeers = append(relayBlockPeers, p)
-				} else {
-					relayHeaderPeers = append(relayHeaderPeers, p)
-				}
+				relayHeaderPeers = append(relayHeaderPeers, p)
 			}
-			go cs.gateway.Broadcast("RelayBlock", currentBlock, relayBlockPeers)
 			go cs.gateway.Broadcast("RelayHeader", currentBlock.Header(), relayHeaderPeers)
```

Comment:
ditto, this can be `go cs.gateway.Broadcast("RelayHeader", currentBlock.Header(), cs.gateway.Peers())`